lib/types/SSI.types.ts

```diff
@@ -0,0 +1,91 @@
+import { PresentationDefinition, PresentationSubmission } from '@sphereon/pe-models';
+
+export interface CredentialSubject {
+  id?: string;
+
+  [x: string]: unknown;
+}
+
+export enum ProofType {
+  Ed25519Signature2018 = 'Ed25519Signature2018',
+  Ed25519Signature2020 = 'Ed25519Signature2020',
+  EcdsaSecp256k1Signature2019 = 'EcdsaSecp256k1Signature2019',
+  EcdsaSecp256k1RecoverySignature2020 = 'EcdsaSecp256k1RecoverySignature2020',
+  JsonWebSignature2020 = 'JsonWebSignature2020',
+  RsaSignature2018 = 'RsaSignature2018',
+  GpgSignature2020 = 'GpgSignature2020',
+  JcsEd25519Signature2020 = 'JcsEd25519Signature2020',
+  BbsBlsSignatureProof2020 = 'BbsBlsSignatureProof2020',
+  BbsBlsBoundSignatureProof2020 = 'BbsBlsBoundSignatureProof2020',
+}
+
+export enum ProofPurpose {
+  assertionMethod = 'assertionMethod',
+  authentication = 'authentication',
+  keyAgreement = 'keyAgreement',
+  contractAgreement = 'contactAgreement',
+  capabilityInvocation = 'capabilityInvocation',
+  capabilityDelegation = 'capabilityDelegation',
+}
+
+export interface Proof {
+  type: ProofType | string; // The proof type
+  created: string; // The ISO8601 date-time string for creation
+  proofPurpose: ProofPurpose | string; // The specific intent for the proof
```

Comment:
I think it's a good idea to limit user options here. when docs a little stable, we can reconsider this

Reply:
Not sure what you mean. There are values in the spec, but it also explicitly allows for future extensions.  This allows for that